packages/application-shell/src/components/service-page-project-switcher/service-page-project-switcher.spec.js

```diff
@@ -1,81 +1,59 @@
-        match={{ params: { projectKey: 'test-1' } }}
-      />
-    );
-  });
-  it('should render <FetchUser>', () => {
-    expect(wrapper).toRender(FetchUser);
-  });
-  describe('when user is loading', () => {
-    beforeEach(() => {
-      fetchUserChildrenWrapper = shallow(
-        <div>
-          {wrapper.find(FetchUser).prop('children')({
-            isLoading: true,
-          })}
-        </div>
+  describe('when user has access to no projects', () => {
+    it('should render nothing', async () => {
+      const rendered = renderApp(
+        <Route path={`/:projectKey`} component={ServicePageProjectSwitcher} />,
+        {
+          route: '/test-1',
+          user: {
+            projects: {
+              total: 0,
+              results: [],
+            },
+          },
+        }
       );
-    });
-    it('should render null', () => {
-      expect(fetchUserChildrenWrapper).toContainReact(<div />);
-    });
-  });
-  describe('when user is not loading', () => {
-    describe('when user has no projects', () => {
-      beforeEach(() => {
-        fetchUserChildrenWrapper = shallow(
-          <div>
-            {wrapper.find(FetchUser).prop('children')({
-              isLoading: false,
-              user: { projects: { total: 0 } },
-            })}
-          </div>
-        );
-      });
-      it('should render null', () => {
-        expect(fetchUserChildrenWrapper).toContainReact(<div />);
+      await wait(() => {
+        expect(
+          rendered.container.querySelector('[name="project-switcher"]')
+        ).not.toBeInTheDocument();
       });
     });
-    describe('when user has projects', () => {
-      let projectSwitcherWrapper;
-      describe('<ProjectSwitcher>', () => {
-        beforeEach(() => {
-          projectSwitcherWrapper = shallow(
-            <div>
-              {wrapper.find(FetchUser).prop('children')({
-                isLoading: false,
-                user: { projects: { total: 1 } },
-              })}
-            </div>
-          );
-        });
-        it('should render <ProjectSwitcher>', () => {
-          expect(projectSwitcherWrapper).toRender(ProjectSwitcher);
-        });
-        it('should pass projectKey to <ProjectSwitcher>', () => {
-          expect(projectSwitcherWrapper.find(ProjectSwitcher)).toHaveProp(
-            'projectKey',
-            'test-1'
-          );
-        });
-        it('should pass total to <ProjectSwitcher>', () => {
-          expect(projectSwitcherWrapper.find(ProjectSwitcher)).toHaveProp(
-            'total',
-            1
-          );
-        });
-      });
+  });
+  describe('when user has access to some projects', () => {
+    it('should render dropdown', async () => {
+      const mockedRequest = [
+        createGraphqlResponseForProjectsQuery({
+          numberOfProjects: 1,
+        }),
+      ];
+      const rendered = renderApp(
+        <Route path={`/:projectKey`} component={ServicePageProjectSwitcher} />,
```

Comment:
This pattern will be deprecated in react-router. I think it would make sense to just render it as children.

```js
  <Route path={`/:projectKey`}><ServicePageProjectSwitcher /></Route>
```

Reply:
True, I'll change that.